arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3279,13 +3354,18 @@ Result ClusterInfo::ensureIndexCoordinatorInner(LogicalCollection const& collect
             AgencyPrecondition(planIndexesKey, AgencyPrecondition::Type::EMPTY, false));
         IndexId indexId{arangodb::basics::StringUtils::uint64(
             newIndexBuilder.slice().get("id").copyString())};
-        if (!_agency.sendTransactionWithFailover(trx, 0.0).successful()) {
+        result = _agency.sendTransactionWithFailover(trx, 0.0);
+        if (!result.successful()) {
           // We just log the problem and move on, the Supervision will repair
           // things in due course:
           LOG_TOPIC("d9420", INFO, Logger::CLUSTER)
               << "Could not remove isBuilding flag in new index "
               << indexId.id() << ", this will be repaired automatically.";
         }
+        auto& cache = _server.getFeature<ClusterFeature>().agencyCache();
```

Comment:
Let's put this in an else, such that it is only done if the write was successful. Otherwise it might throw.